cli/azd/pkg/pipeline/azdo_provider.go

```diff
@@ -418,56 +418,61 @@ func (p *AzdoScmProvider) promptForAzdoRepository(ctx context.Context, console i
 	return remoteUrl, nil
 }
 
-// defines the structure of an ssl git remote
-var azdoRemoteGitUrlRegex = regexp.MustCompile(`^git@ssh.dev.azure\.com:(.*?)(?:\.git)?$`)
-
-// defines the structure of an HTTPS git remote
-var azdoRemoteHttpsUrlRegex = regexp.MustCompile(`^https://[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*:*.+@dev.azure\.com/(.*?)$`)
-
 // ErrRemoteHostIsNotAzDo the error used when a non Azure DevOps remote is found
 var ErrRemoteHostIsNotAzDo = errors.New("existing remote is not an Azure DevOps host")
 
 // ErrSSHNotSupported the error used when ssh git remote is detected
 var ErrSSHNotSupported = errors.New("ssh git remote is not supported. " +
 	"Use HTTPS git remote to connect the remote repository")
 
-// helper function to determine if the provided remoteUrl is an azure devops repo.
-// currently supports AzDo PaaS
-func isAzDoRemote(remoteUrl string) error {
-	if azdoRemoteGitUrlRegex.MatchString(remoteUrl) {
-		return ErrSSHNotSupported
-	}
-	slug := ""
-	for _, r := range []*regexp.Regexp{azdoRemoteGitUrlRegex, azdoRemoteHttpsUrlRegex} {
-		captures := r.FindStringSubmatch(remoteUrl)
-		if captures != nil {
-			slug = captures[1]
-		}
-	}
-	if slug == "" {
-		return ErrRemoteHostIsNotAzDo
-	}
-	return nil
+type azdoRemote struct {
+	Project        string
+	RepositoryName string
 }
 
-func parseAzDoRemote(remoteUrl string) (string, error) {
-	for _, r := range []*regexp.Regexp{azdoRemoteGitUrlRegex, azdoRemoteHttpsUrlRegex} {
-		captures := r.FindStringSubmatch(remoteUrl)
-		if captures != nil {
-			return captures[1], nil
-		}
+var httpsRegex = regexp.MustCompile(
+	//nolint:lll
+	`^https:\/\/(?:.*)?(dev\.azure\.com|visualstudio\.com)\/([a-zA-Z0-9]+)\/([a-zA-Z0-9]+)\/_git\/([a-zA-Z0-9]+)$`)
```

Comment:
I'm seeing both

`https://sanitizedorg.visualstudio.com/DefaultCollection/sanitizedrepo/_git` (grabbed from real ado project)

and also

`https://sanitizedorg.visualstudio.com/sanitizedprojct/_git`

so the "project collection" (group of projects) is optional and needs to be handled.